packages/EasyCodingStandard/packages/SniffRunner/tests/Application/FixerTest.php

```diff
@@ -28,7 +28,7 @@ protected function setUp(): void
         $fileInfo = new SplFileInfo(__DIR__ . '/FixerSource/SomeFile.php', 'FixerSource', 'FixerSource/SomeFile.php');
 
         $this->file = $fileFactory->createFromFileInfo($fileInfo);
-        $this->fixer = $this->container->get(Fixer::class);
+        $this->fixer = clone $this->container->get(Fixer::class);
```

Comment:
Magic? Try to avoid this

Reply:
There is no better way. Fixer needs to have internal state reset after each test

Reply:
Then reset it

Reply:
It's not cloning container, but Fixer

Reply:
Cloning services is like having public properties. You lose control over the state

Reply:
Well this is a workaround for magic you already put in place. You cache symfony container in static property which is persisted between tests and makes tests state sensitive. I don't clone it for no reason. Alright I will reset whole container if you believe that's better